File: src/_118/kotlin/Solution.kt
package _118.kotlin

/**
 * @author relish
 * @since 2018/05/03
 */
class Solution {
    fun generate(numRows: Int): List<List<Int>> {
        val list = ArrayList<ArrayList<Int>>()
        for (i in 0 until numRows) {
            val item = ArrayList<Int>()
            item.add(1)
            if (i > 1) {
                for (j in 1 until i) {
                    val prev = list[i - 1]
                    item.add(prev[j - 1] + prev[j])
                }
            }
            if (i > 0) item.add(1)
            list.add(item)
        }
        return list
    }
}

fun main(args: Array<String>) {
    val message = Solution().generate(10)
    for(item in message){
        println(item)
    }
}
//2 3 4 5 6 7 8
//3 6 10 15 21 28

File: src/_118/Solution.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
  let arr = []
  for(var i = 1; i <= numRows; i++) {
      if(i === 1) {
          arr.push([1])
          continue
      }
      if(i === 2) {
          arr.push([1, 1])
          continue
      }
      let innerArr = [1]
      for(var j = 2; j < i; j++) {
          innerArr.push(arr[i-2][j-2] + arr[i-2][j-1])
      }
      innerArr.push(1)
      arr.push(innerArr)
  }
  return arr
};